x-pack/plugins/ml/common/util/query_utils.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { cloneDeep } from 'lodash';
+import { isPopulatedObject } from './object_utils';
+
+export const getQueryExcludingFrozen = (originalQuery: QueryDslQueryContainer | undefined) => {
```

Comment:
nit pick, the name `getQueryExcludingFrozen` isn't obvious to me that this function will add a new condition to the supplied query.
It's a difficult one to name, but maybe something like `addExcludeFrozenToQuery` would be clearer

Reply:
Updated here https://github.com/elastic/kibana/pull/122306/commits/89ffc85abace99972b5c36a5508ae569591ac9cb